lib/logic/plan_instance_cubit.dart

```diff
@@ -31,23 +34,25 @@ class PlanInstanceCubit extends ReloadableCubit {
 		var completedTasks = await _dataRepository.getCompletedTaskCount(planInstance.id);
 		var uiPlanInstance = UIPlanInstance.fromDBModel(planInstance, plan.name, completedTasks, elapsedTime, getDescription(plan, planInstance.date));
 		var allTasksInstances = await _dataRepository.getTaskInstances(planInstanceId: _planInstanceId);
+		bool isOtherPlanInProgress = planInstance.state == PlanInstanceState.active ? false : await _activeTaskService.isAnyTaskActive(childId: planInstance.assignedTo);
 
 		List<UITaskInstance> uiInstances = await _taskInstancesService.mapToUIModels(allTasksInstances);
-		emit(ChildTasksLoadSuccess(uiInstances, uiPlanInstance));
+		emit(ChildTasksLoadSuccess(uiInstances, uiPlanInstance, isOtherPlanInProgress));
 	}
 }
 
 class ChildTasksLoadSuccess extends DataLoadSuccess {
 	final List<UITaskInstance> tasks;
 	final UIPlanInstance planInstance;
+	final bool isOtherPlanInProgress;
```

Comment:
I left flags because they are good information for typical situations. I added Dialog as you suggested.

Reply:
Ok, I removed local flags because they were causing too many problems...
Everytime we want to start a task there's a query checking if any task is in progress and if it is in progress then we check if we are running this task.